File: app/src/main/java/org/stepic/droid/core/presenters/contracts/LoadCourseView.java
package org.stepic.droid.core.presenters.contracts;

import org.stepic.droid.events.courses.CourseCantLoadEvent;
import org.stepic.droid.events.courses.CourseFoundEvent;
import org.stepic.droid.events.courses.CourseUnavailableForUserEvent;

public interface LoadCourseView {
    void onCourseFound(CourseFoundEvent event);

    void onCourseUnavailable(CourseUnavailableForUserEvent event);

    void onInternetFailWhenCourseIsTriedToLoad(CourseCantLoadEvent event);
}

File: app/src/main/java/org/stepic/droid/core/presenters/contracts/LoadCourseView.kt
package org.stepic.droid.core.presenters.contracts

import org.stepik.android.model.Course

interface LoadCourseView {
    fun onCourseFound(course: Course)

    fun onCourseUnavailable(courseId : Long)

    fun onInternetFailWhenCourseIsTriedToLoad()
}
